fe/fe-core/src/main/java/com/starrocks/catalog/JDBCTable.java

```diff
@@ -180,9 +182,34 @@ private static String buildCatalogDriveName(String uri) {
         // -> jdbc_postgresql_172.26.194.237_5432_db_pg_select
         // requirement: it should be used as local path.
         // and there is no ':' in it to avoid be parsed into non-local filesystem.
-        return uri.replace("//", "")
-                .replace("/", "_")
-                .replace(":", "_");
+
+        String ans = uri.replace("//", "")
+                .replace('/', '_')
+                .replace(':', '_');
+
+        // currently we use this uri as part of name of download file.
+        // so if this uri is too long, we might fail to write file on BE side.
+        // so here we have to shorten it to reduce fail probability because of long file name.
+
+        // in most cases, jdbc uri is not that long, so we can preserve original information.
+        if (ans.length() > 128) {
```

Comment:
might be to make it configurable in case in user's env, it needs to be even shorter.

Reply:
well,  I've counted that 128 will be very safe without filename too long problem.

if you insist, I can add a config item.

Reply:
can't hold high expectation on user's env. e.g what kind of fs type they will use. lol

Reply:
@kevincai do you insist or just suggestion? if suggestion, I think we can introduce config item later. because by my calculation, 128 bytes is enough.

Reply:
ok. do it later or encounter it in user's env.